gstd/gstd_http.c

```diff
@@ -0,0 +1,406 @@
+        "Wrong query param provided, \"name\" doesn't exist");
+    goto out;
+  }
+  if (!description) {
+    ret = GSTD_BAD_VALUE;
+    GST_ERROR_OBJECT (session,
+        "Wrong query param provided, \"description\" doesn't exist");
+    goto out;
+  }
+
+  message = g_strdup_printf ("create %s %s %s", path, name, description);
+  ret = gstd_parser_parse_cmd (session, message, output);
+  g_free (message);
+
+out:
+  return ret;
+}
+
+static GstdReturnCode
+do_put (SoupServer * server, SoupMessage * msg, char *name, char **output,
+    const char *path, GstdSession * session)
+{
+  gchar *message = NULL;
+  GstdReturnCode ret = GSTD_EOK;
+
+  g_return_val_if_fail (server, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (msg, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (session, GSTD_NULL_ARGUMENT);
+
+  if (!name) {
+    ret = GSTD_BAD_VALUE;
+    GST_ERROR_OBJECT (session,
+        "Wrong query param provided, \"name\" doesn't exist");
+    goto out;
+  }
+
+  message = g_strdup_printf ("update %s %s", path, name);
+  ret = gstd_parser_parse_cmd (session, message, output);
+  g_free (message);
+
+out:
+  return ret;
+}
+
+static GstdReturnCode
+do_delete (SoupServer * server, SoupMessage * msg, char *name,
+    char **output, const char *path, GstdSession * session)
+{
+  gchar *message = NULL;
+  GstdReturnCode ret = GSTD_EOK;
+
+  g_return_val_if_fail (server, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (msg, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (session, GSTD_NULL_ARGUMENT);
+
+  if (!name) {
+    ret = GSTD_BAD_VALUE;
+    GST_ERROR_OBJECT (session,
+        "Wrong query param provided, \"name\" doesn't exist");
+    goto out;
+  }
+
+  message = g_strdup_printf ("delete %s %s", path, name);
+  ret = gstd_parser_parse_cmd (session, message, output);
+  g_free (message);
+
+out:
+  return ret;
+}
+
+static void
+do_request (SoupServer * server, SoupMessage * msg, GHashTable * query,
+    GstdSession * session, const char *path)
+{
+  gchar *response = NULL;
+  gchar *name = NULL;
+  gchar *description_pipe = NULL;
+  GstdReturnCode ret = GSTD_BAD_COMMAND;
+  gchar *output = NULL;
+  const gchar *description = NULL;
+  SoupStatus status = SOUP_STATUS_OK;
+
+  g_return_if_fail (server);
+  g_return_if_fail (msg);
+  g_return_if_fail (session);
+
+  if (query != NULL) {
+    name = g_hash_table_lookup (query, "name");
+    description_pipe = g_hash_table_lookup (query, "description");
+  }
+
+  if (msg->method == SOUP_METHOD_GET) {
+    ret = do_get (server, msg, &output, path, session);
+  } else if (msg->method == SOUP_METHOD_POST) {
+    ret = do_post (server, msg, name, description_pipe, &output, path, session);
+  } else if (msg->method == SOUP_METHOD_PUT) {
+    ret = do_put (server, msg, name, &output, path, session);
+  } else if (msg->method == SOUP_METHOD_DELETE) {
+    ret = do_delete (server, msg, name, &output, path, session);
+  } else {
+    soup_message_set_status (msg, SOUP_STATUS_NOT_IMPLEMENTED);
```

Comment:
This case is wrong. If we fall here, we will return a successful message, while we shouldn't. Please double check.

Reply:
I will remove the else, and return not found if the verb is not correct.